base/sparse/sparsematrix.jl

```diff
@@ -164,7 +164,7 @@ end
 
 function convert{Tv,Ti}(::Type{SparseMatrixCSC{Tv,Ti}}, M::AbstractMatrix)
     m, n = size(M)
-    (I, J, V) = findnz(M)
+    (I, J, V) = isempty(M) ? (Ti[],Ti[],Tv[]) : findnz(M)
```

Comment:
It seems like a better fix for this particular issue might be to change `findnz` so that it can work on Any arrays, which I think is just a matter or replacing `zeros` with `Array` [here](https://github.com/JuliaLang/julia/blob/edbd92d420c78518bfa7392e9a1935c86a481618/base/array.jl#L1169). OTOH, SparseMatrixCSC{Any} is kind of broken anyway since trying to access a zero entry will throw the same MethodError for `zero`.